### PR TITLE
CLA StreamsLookup ErrorHandler

### PR DESCRIPTION
**Pull request by thedriftofwords**
https://documentation-git-fork-thedriftofwords-doc-d0569d-chainlinklabs.vercel.app/chainlink-automation/guides/streams-lookup-error-handler

https://documentation-git-fork-thedriftofwords-doc-d0569d-chainlinklabs.vercel.app/chainlink-automation/reference/automation-interfaces#checkerrorhandler-function

### CODE REVIEW

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+}
+```
+
+### Testing checkErrorHandler
+
+`checkErrorHandler` is simulated offchain. When `upkeepNeeded` returns `true`, Automation runs `performUpkeep` onchain using the `performData` from `checkErrorHandler`. If the `checkErrorHandler` function itself reverts, `performUpkeep` does not run.
+
+If you need to force errors in StreamsLookup while testing, you can try the following methods:
+
+- Specifying an incorrect feedID to force error code 800400 (`ErrCodeStreamsBadRequest`)
```
Comment (dwightjl) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:91]:
```suggestion
- Specifying an incorrect `feedID` to force error code 800400 (`ErrCodeStreamsBadRequest`)
```

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+- Specifying a future timestamp to force error code 808206 (where partial content is received) for both single feedID and bulk feedID requests
```
Comment (dwightjl) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:92]:
```suggestion
- Specifying a future timestamp to force error code 808206 (where partial content is received) for both single `feedID` and bulk `feedID` requests
```

File: src/features/chainlink-automation/common/StreamsLookupInterface.mdx [hunk 3]
```diff
@@ -47,3 +47,21 @@ This is a `view` function that will be simulated offchain to receive the signed
 | -------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
 | `performData`  | `bytes`   | Data encoded in the `oracleCallback` function that will be used to execute function onchain, e.g. encode signed reports and log data if need be | `bytes`            |
 | `upkeepNeeded` | `boolean` | When set to `true` this will trigger the `performUpkeep` function to be executed onchain                                                        | `true` of `false`  |
+
+### checkErrorHandler function
+
+This function is simulated offchain to respond to [error codes](/chainlink-automation/guides/streams-lookup-error-handler#error-codes) before sending data onchain via `performUpkeep`.
```
Comment (dwightjl) [inline src/features/chainlink-automation/common/StreamsLookupInterface.mdx:53]:
```suggestion
This function is simulated offchain to respond to [error codes](/chainlink-automation/guides/streams-lookup-error-handler#error-codes) before sending data onchain in the `performUpkeep` function. 
```

File: src/features/chainlink-automation/common/StreamsLookupInterface.mdx [hunk 4]
```diff
@@ -47,3 +47,21 @@ This is a `view` function that will be simulated offchain to receive the signed
...
+
+### checkErrorHandler function
+
+This function is simulated offchain to respond to [error codes](/chainlink-automation/guides/streams-lookup-error-handler#error-codes) before sending data onchain via `performUpkeep`. 
+
+#### Parameters
+
+| Variable Name | Type      | Description                                                    | Permissible Values                                                             |
+| ------------- | --------- | -------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| `errorCode`      | `uint` | Error code returned by StreamsLookup error handler | e.g. "800400" (`ErrCodeStreamsBadRequest`) |
```
Comment (dwightjl) [inline src/features/chainlink-automation/common/StreamsLookupInterface.mdx:59]:
```suggestion
| `errorCode`      | `uint` | Error code returned by StreamsLookup error handler | Example: "800400" (`ErrCodeStreamsBadRequest`) |
```

File: src/features/chainlink-automation/common/StreamsLookupInterface.mdx [hunk 5]
```diff
@@ -47,3 +47,21 @@ This is a `view` function that will be simulated offchain to receive the signed
...
+| `extraData`   | `bytes`   | Bytes data sent as part of original MercuryLookup revert error | e.g. can send log data if you want to decode to use                            |
```
Comment (dwightjl) [inline src/features/chainlink-automation/common/StreamsLookupInterface.mdx:60]:
```suggestion
| `extraData`   | `bytes`   | Bytes data sent as part of the original `MercuryLookup` revert error | This field can send log data if you want to decode it and use it.                            |
```

File: src/features/chainlink-automation/common/StreamsLookupInterface.mdx [hunk 6]
```diff
@@ -47,3 +47,21 @@ This is a `view` function that will be simulated offchain to receive the signed
...
+
+#### Outputs
+
+| Variable Name  | Type      | Description                                                                                                                                     | Permissible Values |
+| -------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
+| `performData`  | `bytes`   | Data encoded in the `oracleCallback` function that will be used to execute function onchain, e.g. encode signed reports and log data if need be | `bytes`            |
+| `upkeepNeeded` | `boolean` | When set to `true` this will trigger the `performUpkeep` function to be executed onchain                                                        | `true` of `false`  |
```
Comment (dwightjl) [inline src/features/chainlink-automation/common/StreamsLookupInterface.mdx:67]:
```suggestion
| `upkeepNeeded` | `boolean` | When set to `true`, this triggers the `performUpkeep` function to be executed onchain                                                        | `true` or `false`  |
```

**Review by dwightjl — CHANGES_REQUESTED** [review]

**Review by dwightjl — APPROVED** [review]

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+    external
+    pure
+    returns (bool, bytes memory)
+{
+    bool success = true;
+    return (true, abi.encode(values, extraData, success));
+}
+
+function checkErrorHandler(
+    bytes calldata errorCode,
```
Comment (khadni) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:204]: Is `errorCode` of type `bytes calldata` or `uint`? In the snippet above, it is a `uint`.

**Review by khadni — CHANGES_REQUESTED** [review]

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+1. Add the `checkErrorHandler` function in your contract to specify how you want to handle [error codes](#error-codes). For example, you could decide to ignore any codes related to bad requests or incorrect input, without running `performUpkeep` onchain:
+
+```solidity
+function checkErrorHandler(
+  uint errorCode,
+  bytes extraData
+) external pure returns (bool upkeepNeeded, bytes memory performData) {
+  // Add custom logic to handle errors offchain here
+  bool upkeepNeeded = true;
```
Comment (khadni) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:49]:
```suggestion
  bool _upkeepNeeded = true;
```

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+  bool success = false;
+  if (errorCode == 800400) {
+    // Bad request error code
+    upkeepNeeded = false;
```
Comment (khadni) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:53]:
```suggestion
    _upkeepNeeded = false;
```

File: src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+  } else {
+    // logic to handle other errors
+  }
+  return (upkeepNeeded, abi.encode(errorCode, extraData, success));
```
Comment (khadni) [inline src/content/chainlink-automation/guides/streams-lookup-error-handler.mdx:57]:
```suggestion
  return (_upkeepNeeded, abi.encode(errorCode, extraData, success));
```

**Review by khadni — CHANGES_REQUESTED** [review]

**Review by dwightjl — APPROVED** [review]

**Pull request merged**